http-gateways/IPLD_GATEWAY.md

```diff
@@ -0,0 +1,196 @@
+# IPLD Gateway Specification
+
+**Authors**:
+
+- Mauve Signweaver ([@RangerMauve](https://github.com/RangerMauve))
+
+----
+
+**Abstract**
+
+IPLD Gateway is an extension of [PATH_GATEWAY.md](./PATH_GATEWAY.md) that enables lower level interaction with IPLD data structures under a specific path.
+
+This document describes the delta between [PATH_GATEWAY.md](./PATH_GATEWAY.md) and this gateway type.
+
+Summary:
+
+- Adds a new `/ipld/{cid}[/{segments}][?{params}]` subpath to the gateway
+- Defines a specification for parsing out extra parameters for individual path segments.
+- Describes how to map to and from `ipld://` URLs
+
+## HTTP API
+
+### `GET /ipld/{cid}[/{segments}][?{params}]`
+
+Resolve IPLD paths to some data.
+
+The path segments will be traversed with any parameters used to transform data along the way.
+
+The `format` query string parameter, or the `Accept` request header can be used to control the format which will be used to return the data.
+
+By default, data will be returned as `dag-json`
+
+<!--
+TODO: Cache control semantics?
+TODO: Add more details on how the traversal works?
+-->
+
+### `HEAD /ipld/{cid}[/{segments}][?{params}]`
+
+Resolves IPLD paths, and yields the same status code and headers as `GET`.
+
+### `POST /ipld/` and `POST /ipld/localhost/`
+
+Upload raw data to IPLD.
+The `body` of the request shall be parsed according to the `Content-Type` as IPLD data via standard encodings.
+`/localhost/` is used to support `POST ipld://localhost/` for uploading IPLD data to local nodes in web browsers that support it.
+
+The response will contain an `ipld://{cid}/` URL pointing at your data.
+
+<!--
+TODO: Only allow `/localhost/`? Get rid of `/localhost` from the spec if light clients with protocol handlers don't matter/
```

Comment:
Do we have use cases where things other than `localhost` could be used in the future? 
e.g. do we want to support POST to IPNS identifier?

Reply:
I've been using `POST` to `ipfs://localhost`, or a `PUT ipfs://cid/` as well as `POST ipns://key` to update CIDs, or `PUT ipns://key` in the [Agregore IPFS Daemon Spec](https://github.com/AgregoreWeb/agregore-ipfs-daemon/blob/main/spec.md)